### ISSUE TITLE
Annotation formatting conflicts with IntelliJ IDEA / Android Studio

### ISSUE THREAD

**Issue by PaulWoitaschek** [pinterest/ktlint#1639]
Field annotations are formatted by AS like this:

```kotlin
package mypackage

private annotation class A
private annotation class B

private class C {
  @field: [
  A
  B
  ]
  val a: Int = 42
}
```

Is reformatted by ktlint to:
```kotlin
package mypackage

private annotation class A
private annotation class B

private class C {
  @field: [
    A
    B
  ]
  val a: Int = 42
}
```

It would be great if ktlint was consistent with Android Studios formatting.

* ktlint: 0.47.1 (CLI)
* Android Studio Dolphin | 2021.3.1 RC 1
Editorconfig:
```editorconfig
root = true

[*]
end_of_line = lf
indent_size = 2
indent_style = space
insert_final_newline = true
max_line_length = 140
trim_trailing_whitespace = true
# noinspection EditorConfigKeyCorrectness
ktlint_disabled_rules = import-ordering, package-name, filename
ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```

**Comment by paul-dingemans** [comment]
Reproduced with IntelliJ IDEA as well.

**Comment by paul-dingemans** [comment]
I believe it would be better if this was fixed in the IntelliJ IDEA formatter for which I have created https://youtrack.jetbrains.com/issue/KTIJ-23398/Inconsistent-formatting-of-arrays.

For consistencies reason, I will (temporarily) align with current IntellI IDEA default formatting.

**Comment by paul-dingemans** [comment]
Closed by #1682
